axes/handlers/database.py

```diff
@@ -119,58 +119,55 @@ def user_login_failed(
+                    "http_accept": request.axes_http_accept,
+                    "path_info":  request.axes_path_info,
+                    "failures_since_start": failures_since_start,
+                    "attempt_time":  request.axes_attempt_time                    
+                }
             )
             # Update failed attempt information but do not touch the username, IP address, or user agent fields,
             # because attackers can request the site with multiple different configurations
             # in order to bypass the defense mechanisms that are used by the site.
 
-            log.warning(
-                "AXES: Repeated login failure by %s. Count = %d of %d. Updating existing record in the database.",
-                client_str,
-                attempt.failures_since_start,
-                get_failure_limit(request, credentials),
-            )
-
             separator = "\n---------\n"
 
-            attempt.get_data = Concat("get_data", Value(separator + get_data))
-            attempt.post_data = Concat("post_data", Value(separator + post_data))
-            attempt.http_accept = request.axes_http_accept
-            attempt.path_info = request.axes_path_info
-            attempt.failures_since_start += 1
-            attempt.attempt_time = request.axes_attempt_time
-            attempt.save()
-        except AccessAttempt.DoesNotExist:
             # Record failed attempt with all the relevant information.
             # Filtering based on username, IP address and user agent handled elsewhere,
             # and this handler just records the available information for further use.
-
-            if not (settings.AXES_ONLY_USER_FAILURES and username is None):
+            if created:
                 log.warning(
                     "AXES: New login failure by %s. Creating new record in the database.",
                     client_str,
                 )
-                AccessAttempt.objects.create(
-                    username=username,
-                    ip_address=request.axes_ip_address,
-                    user_agent=request.axes_user_agent,
-                    get_data=get_data,
-                    post_data=post_data,
-                    http_accept=request.axes_http_accept,
-                    path_info=request.axes_path_info,
-                    failures_since_start=1,
-                    attempt_time=request.axes_attempt_time,
-                )
             else:
+                attempt.get_data = Concat("get_data", Value(separator + get_data))
+                attempt.post_data = Concat("post_data", Value(separator + post_data))
+                attempt.http_accept = request.axes_http_accept
+                attempt.path_info = request.axes_path_info
+                attempt.failures_since_start += 1
+                attempt.attempt_time = request.axes_attempt_time
+                attempt.save()
+                
                 log.warning(
```

Comment:
Logging is moved from "before the save" to "after the save". Could you please keep the logging for the repeat failure before attempting save? If save fails, no log message is emitted.